src/util/StreamerMode.cpp

```diff
@@ -32,51 +42,109 @@ bool isInStreamerMode()
-        {
-            QString processName = QString::fromUtf16(
-                reinterpret_cast<char16_t *>(pWPIs[i].pProcessName));
+            // Fallback to true and showing a warning
 
-            if (broadcastingBinaries().contains(processName))
+            if (shouldShowWarning)
             {
-                cache = true;
-                return true;
+                shouldShowWarning = false;
+
+                if (auto selected = getApp()
+                                        ->windows->getMainWindow()
+                                        .getNotebook()
+                                        .getSelectedPage())
+                {
+                    if (auto container =
+                            dynamic_cast<SplitContainer *>(selected))
+                    {
+                        for (auto &&split : container->getSplits())
+                        {
+                            if (auto channel = split->getChannel();
+                                !channel->isEmpty())
+                            {
+                                channel->addMessage(makeSystemMessage(
+                                    "Streamer mode got enabled, due to pgrep "
+                                    "missing. Install it to fix the issue."));
+                            }
+                        }
+                    }
+                }
             }
-        }
-    }
 
-    if (pWPIs)
-    {
-        WTSFreeMemory(pWPIs);
-    }
+            qCWarning(chatterinoStreamerMode) << "pgrep execution timed out!";
 
-    cache = false;
+            cache = true;
+            return true;
 #endif
+
+#ifdef USEWINSDK
+            if (!IsWindowsVistaOrGreater())
+            {
+                return false;
+            }
+            static bool cache = false;
+            static QDateTime time = QDateTime();
+
+            if (time.isValid() &&
+                time.addSecs(cooldownInS) > QDateTime::currentDateTime())
+            {
+                return cache;
+            }
+            time = QDateTime::currentDateTime();
+
+            WTS_PROCESS_INFO *pWPIs = nullptr;
+            DWORD dwProcCount = 0;
+
+            if (WTSEnumerateProcesses(WTS_CURRENT_SERVER_HANDLE, NULL, 1,
+                                      &pWPIs, &dwProcCount))
+            {
+                //Go through all processes retrieved
+                for (DWORD i = 0; i < dwProcCount; i++)
+                {
+                    QString processName = QString::fromUtf16(
+                        reinterpret_cast<char16_t *>(pWPIs[i].pProcessName));
+
+                    if (broadcastingBinaries().contains(processName))
+                    {
+                        cache = true;
+                        return true;
+                    }
+                }
+            }
+
+            if (pWPIs)
+            {
+                WTSFreeMemory(pWPIs);
+            }
+
+            cache = false;
+#endif
+            return true;
```

Comment:
The default return value has always been false - changing this to true should not be part of the same PR that adds discovery to Linux